cmds/rm/rm.go

```diff
@@ -11,81 +11,131 @@
 //     -i: interactive mode
 //     -v: verbose mode
 //     -R: remove file hierarchies
-//     -r: equivalent to -R
 package main
 
 import (
 	"bufio"
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
+	"syscall"
+	//"io"
 )
 
-var (
-	flags struct {
-		r bool
-		v bool
-		i bool
-	}
-	cmd = "rm [-Rrvi] file..."
-)
-
-func init() {
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
-	}
-	flag.BoolVar(&flags.i, "i", false, "Interactive mode.")
-	flag.BoolVar(&flags.v, "v", false, "Verbose mode.")
-	flag.BoolVar(&flags.r, "R", false, "Remove file hierarchies")
-	flag.BoolVar(&flags.r, "r", false, "Equivalent to -R.")
-	flag.Parse()
+// You can add more flags to this struct
+type rmFlags struct {
+	recursive   bool
+	verbose     bool
+	interactive bool
+	//force       bool
 }
 
-func rm(files []string) error {
-	f := os.Remove
-	if flags.r {
-		f = os.RemoveAll
-	}
-
-	workingPath, err := os.Getwd()
+func recursiveDelete(file string, flags rmFlags) error {
+	input := bufio.NewScanner(os.Stdin)
+	statval, err := os.Stat(file)
 	if err != nil {
 		return err
 	}
+	if statval.IsDir() {
+		//Throws error if flag is not recursive
+		if !flags.recursive {
+			//because it is a path error, the print statement is different and cannot be returned
+			newError := os.PathError{Op: "\nrm:", Path: file, Err: syscall.EISDIR}
+			fmt.Fprintf(os.Stderr, "%v\n", newError.Error())
+			return nil
+		}
+		//At this point, the recursive flag is on
+		if flags.interactive || flags.verbose {
```

Comment:
It seems to me you don't need this if statement if you reorder the code a little bit.

```go
if !flags.interactive && !flags.verbose {
  os.RemoveAll(file)
}

// Must be interactive or verbose now.
fileList, err := ioutil.ReadDir(file)
...
```

This let's you indent less.